install.sh

```diff
@@ -26,11 +26,13 @@ get_password () {
   local password
   local prompt="${1}: "
   local retry_prompt="
-The password cannot be empty!
+The password not be empty and must contain at least 1 uppercase, lowercase, number and special characters.
 ${prompt}"
   read -rs -p "${prompt}" password
-  until [ -n "${password}" ]; do
+  echo "$password" | grep "[A-Z]" | grep "[a-z]" | grep "[0-9]" | grep "[@#$%^&*]"
+  while [ -z "${password}" ] || [ $? -ne 0 ]; do
```

Comment:
@mathemancer The conditions don't work correctly any idea?

Reply:
A few things:
- Password rules like this are a mistake in the first place.
- `[ -z "${password}" ]` overrides `[ $? -ne 0 ]`
  - If the password is missing some required character, but is non-zero length, the `$?` in the second check is 1 because `[ -z "${password}" ] returns 1

Reply:
Actually "overrides" is imprecise. The `[ $? -ne 0 ]` actually just negates whatever is put out by `[ -z "${password}" ]`